discord_slash/client.py

```diff
@@ -349,15 +337,23 @@ def add_slash_command(self,
         """
         name = name or cmd.__name__
         name = name.lower()
-        guild_ids = guild_ids if guild_ids else []
-        if name in self.commands:
-            tgt = self.commands[name]
-            if not tgt.has_subcommands:
-                raise error.DuplicateCommand(name)
-            has_subcommands = tgt.has_subcommands
-            for x in tgt.allowed_guild_ids:
-                if x not in guild_ids:
-                    guild_ids.append(x)
+        guild_ids = guild_ids if guild_ids else ["global"]
+
+        for i in range(len(guild_ids)):
+            if guild_ids[i] is None:
+                guild_ids[i] = "global"
+
+            if guild_ids[i] not in self.commands:
+                self.commands[guild_ids[i]] = {}
+            if name in self.commands[guild_ids[i]]:
+                # prevent duplicate commands in guild
+                tgt = self.commands[guild_ids[i]][name]
+                if not tgt.has_subcommands:
+                    raise error.DuplicateCommand(name)
+                has_subcommands = tgt.has_subcommands
+                for x in tgt.allowed_guild_ids:
+                    if x not in guild_ids:
+                        guild_ids.append(x)
```

Comment:
In this case, I'm mirroring the original code, I simply kept the original code as it seemed to be its intent

Reply:
What about removing the `guild_ids` attribute?

Reply:
aye, certainly could remove it, but im trying to minimise the scope of the changes i make. Right now, a user could update their code to use this lib without issues (excld `slash.commads`). Removing `guild_ids` would be more breaking which im trying to avoid